app/src/androidTest/java/ingsw/group1/findmyphone/LogFragmentTest.java

```diff
@@ -0,0 +1,146 @@
+
+    private static final RandomSMSPeerGenerator GENERATOR = new RandomSMSPeerGenerator();
+    private static final String DB_NAME = "test-db";
+
+    private LogFragment fragment;
+
+    private List<SMSLogEvent> exampleEvents = Arrays.asList(
+            new SMSLogEvent(
+                    LogEventType.RING_REQUEST_SENT,
+                    new SMSContact(
+                            GENERATOR.generateValidPeer(),
+                            "User 1"
+                    ),
+                    System.currentTimeMillis(),
+                    String.valueOf(1000)
+            ),
+            new SMSLogEvent(
+                    LogEventType.RING_REQUEST_RECEIVED,
+                    new SMSContact(
+                            GENERATOR.generateValidPeer(),
+                            "User 2"
+                    ),
+                    System.currentTimeMillis(),
+                    String.valueOf(1000)
+            ),
+            new SMSLogEvent(
+                    LogEventType.LOCATION_REQUEST_RECEIVED,
+                    new SMSContact(
+                            GENERATOR.generateValidPeer(),
+                            "User 3"
+                    ),
+                    System.currentTimeMillis(),
+                    new GeoPosition(
+                            100, 100
+                    ).toString()
+            ),
+            new SMSLogEvent(
+                    LogEventType.LOCATION_REQUEST_SENT,
+                    new SMSContact(
+                            GENERATOR.generateValidPeer(),
+                            "User 4"
+                    ),
+                    System.currentTimeMillis(),
+                    new GeoPosition(
+                            100, 100
+                    ).toString()
+            ),
+            //This event should be translated into an item with red text.
+            new SMSLogEvent(
+                    LogEventType.RING_REQUEST_RECEIVED,
+                    new SMSContact(
+                            GENERATOR.generateValidPeer(),
+                            "User 5"
+                    ),
+                    System.currentTimeMillis(),
+                    null
+            ),
+            //This event should be not shown.
+            new SMSLogEvent()
+    );
+
+    public static ViewAction waitFor(final long millis) {
+        return new ViewAction() {
+            @Override
+            public Matcher<View> getConstraints() {
+                return isRoot();
+            }
+
+            @Override
+            public String getDescription() {
+                return "Wait for " + millis + " milliseconds.";
+            }
+
+            @Override
+            public void perform(UiController uiController, final View view) {
+                uiController.loopMainThreadForAtLeast(millis);
+            }
+        };
+    }
+
+    /**
+     * Rule to create an Activity
+     */
+    @Rule
+    public ActivityTestRule<NavHolderActivity> rule =
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://app.codacy.com/manual/RicDeZen/ingsw-group1-findmyphone/pullRequest?prid=4969671)